src/middleware/auth.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+'use strict';
+
+const jwt = require('jwt-simple');
+const wrap = require('co-express');
+const createError = require('http-errors');
+const _ = require('lodash');
+
+const EXPIRED = 'Token expired';
+const VALIDATION_FAILED = 'Token validation failed';
+const NOT_CONNECTED = 'User not connected';
+const NEED_KEY = 'IDP secret for JWT is not set';
+const ALGORITHM = 'RS256';
+const KEY_END_STRING = '-----END PUBLIC KEY-----\n';
+
+/**
+ * This functions takes a string of concatenated PEM public keys and splits
+ * them in an array.
+ * @param {string} keysString - Concatenated PEM public keys
+ * @returns {Array} Separated PEM public keys
+ */
+function retrieveKeys(keysString) {
+  const keys = keysString.split(KEY_END_STRING);
+
+  return _.map(keys, (item, index) => (index + 1 === keys.length) ? item : item + '-----END PUBLIC KEY-----');
```

Comment:
What you think about using a space ` ` for separate your keys ?

This way you can just omit the map step to re-add the split string  on each keys.

Reply:
They keys are actually formatted this way:
https://github.com/transcovo/express-middleware/pull/23/files#diff-c10ff5c54b7316d2bfab939ddbc876ceR3

A demi-colon is more relevant here but year you're right